### PR TITLE
Tests for GitHub.Sync.installation_event/1

### LINKED ISSUE #1373: Write tests for GitHub.Sync.installation_event/1
# Problem

`GitHub.Sync.installation_event/1` is currently completely untested. Since we want to unit test its individual elements to, it should suffice to test each possible outcome of the transaction performed by the tests, as defined by `GitHub.Sync.installation_event_outcome()`

## Subtasks
- [x] write tests for `{:ok, GithubAppInstallation.t()}` outcome
- [ ] write tests for `{:error, :validation_error_on_syncing_installation, Changeset.t()}` outcome
- [x] write tests for `{:error, :multiple_unprocessed_installations_found}` outcome
- [x] write tests for `{:error, :github_api_error_on_syncing_repos, struct}` outcome
- [ ] write tests for `{:error, :validation_error_on_deleting_removed_repos, {list, list}}` outcome
- [ ] write tests for `{:error, :validation_error_on_syncing_existing_repos, {list, list}}` outcome
- [ ] write tests for `{:error, :validation_error_on_marking_installation_processed, Changeset.t()}` outcome
- [ ] write tests for `{:error, :unexpected_transaction_outcome, any}` outcome

### PR DESCRIPTION
**Pull request by zacck-zz**
# What's in this PR?

Adding test coverage for the GitHub.Sync.installation_event/1

## References
Fixes #1373 

Progress on: #

### CODE REVIEW

**Comment by zacck-zz** [comment]
Struggling to test some validation clauses, however, seeking some help testing for the following clauses
```
{:error, :validating_github_issue, Changeset.t()}
{:error, :validating_user, Changeset.t()}
{:error, :multiple_issue_users_match}
```

File: test/lib/code_corps/github/sync/sync_test.exs [hunk 1]
```diff
@@ -24,6 +24,70 @@ defmodule CodeCorps.GitHub.SyncTest do
...
+
+    end)
+  end
+
+  describe "pull_request_event/1 " do
+    @payload load_event_fixture("pull_request_opened")
+
+
+
+  end
```
Comment (begedin) [inline test/lib/code_corps/github/sync/sync_test.exs]: You probably meant to remove this part.

**Review by begedin — CHANGES_REQUESTED** [review]
Just some extra stuff you forgot to remove and then it's good to merge. 👍

File: test/lib/code_corps/github/sync/sync_test.exs (hunk 1 above)
Comment (zacck-zz) [reply test/lib/code_corps/github/sync/sync_test.exs]: haha yes I did

**Comment by zacck-zz** [comment]
Ok done!!

**Comment by zacck-zz** [comment]
@begedin  ok ✂️  done!

**Review by begedin — APPROVED** [review]

**Comment by begedin** [comment]
@zacck As usual, thank you for your time, work and expertise. 👍

**Pull request merged**